lib/workers/repository/init/inherited.ts

```diff
@@ -0,0 +1,94 @@
+import is from '@sindresorhus/is';
+import { mergeChildConfig } from '../../../config';
+import { parseFileConfig } from '../../../config/parse';
+import type { RenovateConfig } from '../../../config/types';
+import { validateConfig } from '../../../config/validation';
+import {
+  CONFIG_INHERIT_NOT_FOUND,
+  CONFIG_INHERIT_PARSE_ERROR,
+  CONFIG_VALIDATION,
+} from '../../../constants/error-messages';
+import { logger } from '../../../logger';
+import { platform } from '../../../modules/platform';
+import * as template from '../../../util/template';
+
+export async function mergeInheritedConfig(
+  config: RenovateConfig,
+): Promise<RenovateConfig> {
+  if (!config.inheritConfig) {
+    return config;
+  }
+  if (
+    !is.string(config.inheritConfigRepoName) ||
+    !is.string(config.inheritConfigFileName)
+  ) {
+    // Should not happen due to config validation
```

Comment:
```suggestion
    // Config validation should prevent this error
```

Maybe something like this is clearer?